### DISCUSSION TITLE
Add extra parameters to an op that takes dynamic input

### DISCUSSION THREAD

**Discussion by dagsir** [dagster-io/dagster#8498] [Q&A]
I need to add extra parameters to an op that takes dynamic input?

I got this code:
```
from dagster import DynamicOut, DynamicOutput, job, op


@op(out=DynamicOut(str))
def output_fruits():
    fruits = ["Mango", "Durian", "Kiwi"]
    for fruit in fruits:
        yield DynamicOutput(fruit, mapping_key=fruit)


@op
def log_single_fruit(context, fruit: str) -&gt; str:
    context.log.info(f"log_single_fruit: {fruit}")
    return fruit


@job
def process_fruits():
    fruit = output_fruits()

    fruit.map(log_single_fruit)
```
It is working, and it is logging each fruit.

However, how do I write my code if I want an extra parameter in the `log_single_fruit` op?
Like:
```def log_single_fruit(context, fruit: str, something: int) -> str:```
I think I need to rewrite the job, but I’m not sure how.

---

*The question was originally asked in [Dagster Slack](https://dagster.slack.com/archives/C01U954MEER/p1638362757230200?thread_ts=1638362757.230200&cid=C01U954MEER).*

**Comment by yuhan** [comment]
You can use a lambda function here:
```python
s = something()
fruit.map(lambda f: log_single_fruit(fruit=f, something=s))
```

Here's the doc: https://docs.dagster.io/concepts/ops-jobs-graphs/dynamic-graphs#additional-arguments
